web/packages/teleport/src/Roles/RoleEditor/RoleEditorAdapter.tsx

```diff
@@ -0,0 +1,240 @@
+  onDelete: () => Promise<void>;
+  onCancel: () => void;
+}) {
+  const theme = useTheme();
+  const [convertAttempt, convertToRole] = useAsync(
+    async (yaml: string): Promise<RoleWithYaml | null> => {
+      if (resources.status === 'creating' || !resources.item) {
+        return null;
+      }
+      return {
+        yaml,
+        object: await yamlService.parse<Role>(YamlSupportedResourceKind.Role, {
+          yaml,
+        }),
+      };
+    }
+  );
+
+  const originalContent = resources.item?.content ?? '';
+  useEffect(() => {
+    convertToRole(originalContent);
+  }, [originalContent]);
+
+  return (
+    <Flex flex="1">
+      <Flex
+        flexDirection="column"
+        borderLeft={1}
+        borderColor={theme.colors.interactive.tonal.neutral[0]}
+        backgroundColor={theme.colors.levels.surface}
+        width="700px"
+      >
+        {convertAttempt.status === 'processing' && (
+          <Flex
+            flexDirection="column"
+            alignItems="center"
+            justifyContent="center"
+            flex="1"
+          >
+            <Indicator />
+          </Flex>
+        )}
+        {convertAttempt.status === 'error' && (
+          <Danger>{convertAttempt.statusText}</Danger>
+        )}
+        {convertAttempt.status === 'success' && (
+          <RoleEditor
+            originalRole={convertAttempt.data}
+            onCancel={onCancel}
+            onSave={onSave}
+            onDelete={onDelete}
+          />
+        )}
+      </Flex>
+      <Flex flex="1" alignItems="center" justifyContent="center" m={3}>
+        <Box>
+          <H1 mb={2}>Teleport Policy</H1>
+          <Flex mb={4}>
+            <Box flex="1" width="min-content">
+              <P>
+                Teleport Policy will visualize resource access paths as you
+                create and edit roles so you can always see what you are
+                granting before you push a role into production.
+              </P>
+            </Box>
+            <Flex flex="0 0 auto" ml={6} alignItems="start">
+              <ButtonLockedFeature noIcon py={0} width={undefined}>
+                Contact Sales
+              </ButtonLockedFeature>
+              <ButtonSecondary
+                as="a"
+                href="https://goteleport.com/platform/policy/"
+                target="_blank"
+                ml={2}
+              >
+                Learn More
+              </ButtonSecondary>
+            </Flex>
+          </Flex>
+          <Flex
+            flexDirection="column"
+            bg={theme.colors.levels.surface}
+            borderRadius={3}
+            // I found hardcoding the width of the container to be the only way
+            // to play well with the StepSlider below. Trying to use implicit
+            // width in combination with `width: min-content` inside a
+            // StepSlider child caused a miscalculation of child's height.
+            width={promoImageWidth + 2 * 2}
```

Comment:
could we just use `+4` in that case?

Reply:
I went with "2*2", because I hoped it makes it more clear that I'm adjusting for 2 borders of 2 pixels.